0000-catalog.md

```diff
@@ -0,0 +1,152 @@
+
+## Permissions
+
+Creation of a Grid Catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field.  An organization must have the GS1
+company prefix in the “gs1_company_prefixes” metadata field for its Pike
+organization.  (Organization-level metadata will need to be implemented in
+Pike.) When a catalog is created, its owning organization is stored with the
+catalog in an “owner” field.
+
+Deletion of a catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field.  A setting will turn off deletion
+entirely, with the intent that it could be enabled/disabled by a Grid
+administrator.  Disabling delete is useful because external systems may have
+references to these products and deleting them could leave dangling references.
+
+Replication of a catalog is restricted to agents acting on behalf of an
+organization. The purpose of replication is to enable agents to duplicate a
+catalog in state with the intention of modifying the catalog for a different
+specification. It reduces the overhead needed to recreate a catalog only to
+add/modify few or many attributes of a catalog, or the products associated.
+
+In the case of **catalog operations** agents from the org they represent will
+need the operational permission to perform the desired operation on a Grid
+Catalog.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The primary object stored in state is **Catalog**, which consists of a
+catalog_id ([UUID](https://crates.io/crates/uuid)), an owner (org_id compatible
+w/Pike), an expiry_date (unix timestamp), an *optional* **Name**, and a list of
+property key/value pairs.  The properties available are defined by the Grid
+Property Schema Transaction Family and are restricted to the fields and rules of
+the GS1 Catalog schema (which will be defined at a later time). Transactions
+which are responsible for setting product state values must ensure that those
+properties conform with the requirements of the GS1 Catalog Property Schema (to
+be defined at a later time).
+
+``` 
+message Catalog { 
+    string catalog_id = 1; 
+    string owner = 2; 
+    string expiry_date = 3; 
+    name = 4; 
+    ProductList products = 5; 
+    repeated PropertyValue properties = 6; 
+} 
+```
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+The design of catalog in this manner leaves it extensible for domain specific
+use. The concept of price has been discussed and it's been decided to include a
+notion of price within products added to a catalog. A simple implmentation of
+price is fine and currently serves as a placeholder, while a more robust
+standard of modeling price data can be implemented from a future pricing-related
+RFC. A GS1 pricing standard that can be leveraged in the writing of such an RFC
+would be [GS1 Price
+Sync](https://www.gs1.org/docs/gdsn/3.1/BMS_Price_Sync_r3p1p3_i1p3p5_23May2017.pdf).
+
```

Comment:
Understanding that pricing is variable by trading partner relationship for the same product, how would this be implemented? A separate record for the same item  - tied to the partner the supplier may be doing business with? Goes back to the question above - is the intent to have a 1:1 catalog for every trading partner relationship or model more what is today which is a 1:many?

Reply:
First question:
From the Rational and Alternatives section: 
> The concept of price has been discussed and it's been decided to include
the notion of price within products added to a catalog. A simple implementation of
price is fine and currently serves as a placeholder, while a more robust
standard of modeling price data can be implemented from a future pricing-related
RFC.

Second question: Yes, we could require something like a trading_partner_id for catalogs of type "TRADING_PARTNER_CATALOG". 

Third question: Addressed above

Reply:
I don't know if it makes sense to sub-type these things. I think we should keep it loose and allow the creating organizations to define what it is. This may be very supplier (catalog-creator) specific. For example, an organization might have a single customer in a given segment and may publish a unique catalog of products directed at that customer. Then, in the future, they might acquire a different customer in that same segment and decide that catalog is suitable for providing to them as well. Ultimately, the segmentation is based on choices the creators make and what parties they choose to share the data with.

Reply:
The flexibility feels really important.

The supplier may have contractual obligations with the trading partner to keep some/all catalog items in a private context (however that might work) - probably more B2B.

Perhaps the supplier having a "flag" that allows catalogs/products/items to be "listed" as B2B, B2C, etc., for traditional commerce allows some "logic" flow to be identified for other commerce types, things like Business2Government.

Reply:
> The flexibility feels really important.
> 
> The supplier may have contractual obligations with the trading partner to keep some/all catalog items in a private context (however that might work) - probably more B2B.
> 
> Perhaps the supplier having a "flag" that allows catalogs/products/items to be "listed" as B2B, B2C, etc., for traditional commerce allows some "logic" flow to be identified for other commerce types, things like Business2Government.

I see where you're coming from. I think in the context of "may" have contractual obligations, a "flag" field for this would be better included as an additional property_value enforced by a grid schema that could be a fork of the catalog_product schema.